patches/native-comp-env-setup.patch

```diff
@@ -0,0 +1,41 @@
+diff --git a/lisp/emacs-lisp/comp.el b/lisp/emacs-lisp/comp.el
+index 25e2de9..6626b69 100644
+--- a/lisp/emacs-lisp/comp.el
++++ b/lisp/emacs-lisp/comp.el
+@@ -2801,6 +2801,36 @@ queued with LOAD %"
+       (comp-run-async-workers)
+       (message "Compilation started."))))
+ 
++;;;###autoload
++(defun native-compile-setup-environment-variables (&rest args)
++  "Ensure LIBRARY_PATH is set correctly when libgccjit is bundled."
++  (when (and (eq system-type 'darwin)
++             (string-match-p "Emacs\.app\/Contents\/MacOS\/?$"
```

Comment:
I usually add the version number to the app name to distinguish 26 from 27 and 28, so the following was necessary to make the patch work:

`(string-match-p "Emacs.*\.app\/Contents\/MacOS\/?$"`

Reply:
Ah, yes that's a good catch. Technically users can rename the app to whatever they want, so I'll change it to simply `"\.app\/Contents\/MacOS\/?$"`.

Reply:
Perfect!